### PR TITLE
feat: Basic `Section` stories

### PR DESCRIPTION
**Pull request by DafyddLlyr**
I spent some time trying to resolve this using Zustand stores without much luck within the Storybook components. I think in order to get this to work we'd need to use a Provider approach, similar to what's outlined here - 

This would be a big refactor and I'm unsure of the real benefit for the effort involved.

I've gone for just a stateless wrapper component to simplify the visual testing of the "Section" component.

### CODE REVIEW

File: editor.planx.uk/src/@planx/components/Section/Section.stories.tsx [hunk 1]
```diff
@@ -14,13 +14,73 @@ type Story = StoryObj<typeof meta>;
...
+      firstSection: {
+        auto: false,
+      },
+    },
   },
 } satisfies Story;
 
-export const WithEditor = () => {
-  return <Wrapper Editor={Editor} Public={Public} />;
-};
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Section/Section.stories.tsx:26]: One downside - I can't get this approach to work with the `Wrapper` component. Not a huge issue for this single component which has a simple Editor representation but it probably shows that this isn't an approach we want to repeat too much.

File: editor.planx.uk/src/@planx/components/Section/Section.stories.tsx [hunk 2]
```diff
@@ -14,13 +14,73 @@ type Story = StoryObj<typeof meta>;
...
+export const SecondSectionCompleted = {
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/Section/Section.stories.tsx:73]: nit: would a `WithNewInformationNeeded` story be more distinct/meaningful to mock than another "completed" tag?

**Review by jessicamcinchak — APPROVED** [review]
Thanks for picking this one up & sorting out !

File: editor.planx.uk/src/@planx/components/Section/Section.stories.tsx (hunk 2 above)
Comment (DafyddLlyr) [reply editor.planx.uk/src/@planx/components/Section/Section.stories.tsx:73]: Good shout - I'll add this.

**Pull request merged**